pkg/adaptor/mongodb/reader.go

```diff
@@ -95,7 +96,10 @@ func (r *Reader) iterateCollection(mgoSession *mgo.Session, in <-chan string, do
 		defer close(out)
 		for {
 			select {
-			case c := <-in:
+			case c, ok := <-in:
+				if !ok {
+					return
+				}
 				log.With("collection", c).Infoln("iterating...")
 				canReissueQuery := r.requeryable(c, mgoSession)
 				var lastID interface{}
```

Comment:
Change above looks OK.

Looking at `lastID`, I don't see where we're updating it. I think it needs to be updated in the `iter.Next` loop below. Wonder if it's possible to write a test that captures the requerying logic.